pkg/util/kubernetes/apiserver/services_test.go

```diff
@@ -0,0 +1,171 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build kubeapiserver
+
+package apiserver
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ericchiang/k8s/api/v1"
+	metav1 "github.com/ericchiang/k8s/apis/meta/v1"
+	"github.com/stretchr/testify/assert"
+)
+
+type podTest struct {
+	ip   string
+	name string
+}
+
+type serviceTest struct {
+	svcName string
+	podIps  []string
+}
+
+func toPtr(str string) *string {
+	return &str
+}
+
+func createSvcList(nodeName string, svcs []serviceTest) v1.EndpointsList {
+	var list v1.EndpointsList
+	for _, svc := range svcs {
+		var endpoints v1.Endpoints
+		var endpointsSubset v1.EndpointSubset
+		endpoints.Metadata = &metav1.ObjectMeta{}
+		endpoints.Subsets = append(endpoints.Subsets, &endpointsSubset)
+		endpoints.Metadata.Name = toPtr(svc.svcName)
+
+		for _, e := range svc.podIps {
+			var ep v1.EndpointAddress
+			ep.NodeName = &nodeName
+			ep.Ip = toPtr(e)
+			endpointsSubset.Addresses = append(endpointsSubset.Addresses, &ep)
+		}
+		list.Items = append(list.Items, &endpoints)
+	}
+	return list
+}
+
+func createPodList(listPodStructs []podTest) v1.PodList {
+	var podlist v1.PodList
+	for _, ps := range listPodStructs {
+		var pod v1.Pod
+		pod.Status = &v1.PodStatus{}
+		pod.Metadata = &metav1.ObjectMeta{}
+		pod.Status.PodIP = toPtr(ps.ip)
+		pod.Metadata.Name = toPtr(ps.name)
+		fmt.Printf("\n ps.name is %s, toPtr ps.name is %q and &ps.name is %q \n\n", ps.name, toPtr(ps.name), &ps.name)
```

Comment:
You should cleanup this line or put it in a proper log

Reply:
🤦‍♂️